tests/Fixer/ControlStructure/NoUnneededControlParenthesesFixerTest.php

```diff
@@ -340,6 +343,47 @@ function foo() { $a = (yield($x)); }
                 $var = clone ($obj1->getSubject() ?? $obj2);
                 ',
             ],
+            [
+                '<?php $stuff += $foo ? $bar() : $baz /* (╯°□°)╯︵ ┻━┻ */ ;',
+                '<?php $stuff += ($foo ? $bar() : $baz) /* (╯°□°)╯︵ ┻━┻ */ ;',
+            ],
+            [
+                '<?php $stuff = $baz ? $this->foo() : $this->bar();',
+                '<?php $stuff = ($baz) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = $this->baz ? $this->foo() : $this->bar();',
+                '<?php $stuff = ($this->baz) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = ($this->baz)() ? $foo["bar"]($baz) : $foo->{$baz}($bar);',
+            ],
+            [
+                '<?php $stuff = ($this->baz)($a) ? $this->foo() : $this->bar();',
+                '<?php $stuff = (($this->baz)($a)) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = $baz($a) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = $baz($a, $b->c->d()) ? $this->foo() : $this->bar();',
+                '<?php $stuff = ($baz($a, $b->c->d())) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = Foo::bar() ? $this->foo() : $this->bar();',
+                '<?php $stuff = (Foo::bar()) ? $this->foo() : $this->bar();',
+            ],
+            [
+                '<?php $stuff = (Foo::bar())($baz) ? $this->foo() : $this->bar();',
+            ],
+            [
+                // Even if it would work without parentheses in this case, let's ensure we keep them for readability.
+                // Unparenthesised ternaries were deprecated in 7.4 and cause fatal error in 8.0+
+                "<?php
+                echo ((1 == 2 ? 'a' : 3 == 4) ? 'b' : 5 == 6) ? 'c' : 'd';
+                echo 1 == 2 ? 'a' : (3 == 4 ? 'b' : (5 == 6 ? 'c' : 'd'));
```

Comment:
This is very tricky, as `? :` has `left-associative prior to PHP 8.0.0` , meaning that reliable checking chained `?` and combinations with `??` do not need `(` `)` is pretty hard.
I wouldn't touch ternaries, but if wanted to support, lets get some cases in like:
```php
<?php
$a =
    (($a ?? $b) ?: ($a ? 1 : 2))
    ??
    (((($a ? $b : $c) ?? $a) ?: $d) ?? $y)
?>
```